src/builder.rs

```diff
@@ -260,6 +260,7 @@ pub struct Builder {
     recipients: Vec<RecipientInfo>,
     flags: Flags,
     anchor: Anchor,
+    assets_burnt: Vec<(AssetId, i64)>,
```

Comment:
`Vec` seems inappropriate since we can have multiple burns for the same asset. I would go with:
```rust
pub struct Builder {
    spends: Vec<SpendInfo>,
    recipients: Vec<RecipientInfo>,
    burn: HashMap<AssetId, ValueSum>,
    flags: Flags,
    anchor: Anchor,
}
```
Order is also important, let's have `flags` and `anchor` at the end.

Reply:
I was thinking its more of a feature that you can burn different amounts separately, but I agree that use case for that will be pretty artificial 

Regarding ValueSum - yeah, actually first I implemented with it and then decided to simplify

Reply:
done